extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs

```diff
@@ -284,60 +287,56 @@ fn extract_fork_config_from_args(db: &dyn SyntaxGroup, attr: &Attribute) -> Opti
         return None;
     }
 
-    let block_id_type = block_id
-        .path(db)
-        .elements(db)
-        .last()
-        .unwrap()
-        .identifier(db)
-        .to_string();
+    let block_id_type = elements[1].clone();
 
     let args = block_id.arguments(db).arguments(db).elements(db);
     let expr = match args.first()?.arg_clause(db) {
         ArgClause::Unnamed(unnamed_arg_clause) => Some(unnamed_arg_clause.value(db)),
         _ => None,
     }?;
-    let block_id = try_get_block_id(db, &block_id_type, &expr)?;
+    let block_id_value = try_get_block_id(db, &block_id_type, &expr)?;
 
-    Some(RawForkConfig::Params(RawForkParams { url, block_id }))
+    Some(ForkConfigString::Params(ForkParamsString {
+        url,
+        block_id_type,
+        block_id_value,
+    }))
 }
 
-fn try_get_block_id(db: &dyn SyntaxGroup, block_id_type: &str, expr: &Expr) -> Option<BlockId> {
+fn try_get_block_id(db: &dyn SyntaxGroup, block_id_type: &str, expr: &Expr) -> Option<String> {
```

Comment:
If we are "parsing" it here anyway, why won't we pass this as enum defined by us? Seems counter-productive to first parse the block id and the return it as two strings anway.

Reply:
That would be basically copy pasting `starknet-rs` structs, not sure if that's what we want. It's not costless to add those structs, additional thing to maintain. I guess that's up to debate, I think both of the solutions are bad

Reply:
Discussed offline, leaving it as it is